apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/pagerules/PageRuleServiceImpl.java

```diff
@@ -817,4 +823,50 @@ public ViewRuleResponse getRuleResponse(Long ruleId) {
                 ruleMetadata.getRuleDescText(), ruleMetadata.getSourceQuestionIdentifier(), sourceValues,
                 ruleMetadata.getLogic(), ruleMetadata.getTargetType(), ruleMetadata.getErrormsgText(), targetValues);
     }
+
+    @Override
+    public Page<ViewRuleResponse> getAllPageRule(Pageable pageRequest) {
+        Page<WaRuleMetadata> ruleMetadataPage =waRuleMetaDataRepository.findAll(pageRequest);
+
+        List<ViewRuleResponse> ruleMetadata =
+                ruleMetadataPage.getContent().stream().map(rule ->new ViewRuleResponse(rule.getId(), rule.getWaTemplateUid(), rule.getRuleCd(),
+                rule.getRuleDescText(), rule.getSourceQuestionIdentifier(), buildSourceTargetValues(rule,true),
+                rule.getLogic(), rule.getTargetType(), rule.getErrormsgText(),
+                        buildSourceTargetValues(rule,false))).toList();
+        return new PageImpl<>(ruleMetadata,ruleMetadataPage.getPageable(),ruleMetadataPage.getTotalElements());
+
+    }
+
+    @Override
+    public Page<ViewRuleResponse> findPageRule(SearchPageRuleRequest request, Pageable pageable) {
+        Page<WaRuleMetadata> ruleMetadataPage =waRuleMetaDataRepository.findAllBySourceValuesContainingIgnoreCaseOrTargetQuestionIdentifierContainingIgnoreCase(request.searchValue(),request.searchValue(),pageable);
+
+        List<ViewRuleResponse> ruleMetadata =
+                ruleMetadataPage.getContent().stream().map(rule ->new ViewRuleResponse(rule.getId(), rule.getWaTemplateUid(), rule.getRuleCd(),
+                        rule.getRuleDescText(), rule.getSourceQuestionIdentifier(), buildSourceTargetValues(rule,true),
+                        rule.getLogic(), rule.getTargetType(), rule.getErrormsgText(),
+                        buildSourceTargetValues(rule,false))).toList();
+        return new PageImpl<>(ruleMetadata,ruleMetadataPage.getPageable(),ruleMetadataPage.getTotalElements());
+    }
+
+    private List<String> buildSourceTargetValues(WaRuleMetadata ruleMetadata, boolean isSource) {
+
+        List<String> sourceValues = new ArrayList<>();
+        List<String> targetValues = new ArrayList<>();
+        if(isSource){
+            if (ruleMetadata.getSourceValues() == null || ruleMetadata.getTargetQuestionIdentifier() == null) {
+                return  sourceValues;
+            } else {
+               return Arrays.stream(ruleMetadata.getSourceValues().split(",")).toList();
```

Comment:
Using `Arrays.asList()` here might be a little better.  The Stream is only being used to create a list so it's just adding overhead.